stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/Sprintf.java

```diff
@@ -95,19 +95,34 @@ public final void execute(final Context context) {
                 }
                 try {
                     char formatSpecifier = format.charAt(j);
+                    BRefType ref = args.getRefValue(k);
                     switch (formatSpecifier) {
                         case 'b':
                         case 'B':
                         case 'd':
                         case 'f':
-                            result.append(String.format("%" + padding + formatSpecifier, args.getRefValue(k).value()));
+                            if (ref != null) {
+                                result.append(String.format("%" + padding + formatSpecifier, ref.value()));
+                            } else {
+                                throw BLangExceptionHelper.getRuntimeException(
+                                        RuntimeErrors.ILLEGAL_FORMAT_CONVERSION,
+                                        format.charAt(j) + " != ()");
+                            }
```

Comment:
Shall we invert the if condition, and fail fast?

Reply:
@anupama-pathirage Isn't it the most probable condition that should be put as the if condition?

Reply:
@Manuri The idea is to get invalid cases out of the way first, either simply exiting or raising exceptions as appropriate. Then add the real code which needs to get executed. With that we don't need else block at all.